src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -164,7 +170,11 @@ def run_python_test(transitive_hydrated_target, pytest, python_setup, source_roo
 
   request = ExecuteProcessRequest(
     argv=(python_binary, './{}'.format(output_pytest_requirements_pex_filename)),
-    env={'PATH': os.pathsep.join(python_setup.interpreter_search_paths)},
+    env={
+      'PATH': text_type(os.pathsep.join(python_setup.interpreter_search_paths)),
```

Comment:
Consider converging on `pants.util.strutil.create_path_env_var()` to do path var construction, as it clearly denotes intent and covers multiple edge cases pretty neatly.